### DISCUSSION TITLE
CreateProcess() failed for executable in repo on Windows

### DISCUSSION THREAD

**Discussion by emoon** [community/community#25948] [Actions]
<p>Hi,</p>
<p>When I try to run a Windows executable in my GH action I get <code>CreateProcess() failed</code> <a href="https://github.com/emoon/HippoPlayer/runs/819601424?check_suite_focus=true" rel="nofollow noopener">https://github.com/emoon/HippoPlayer/runs/819601424?check_suite_focus=true</a> with this log</p>
<pre><code class="lang-auto">  bin\windows\tundra2.exe win64-msvc-debug
  shell: C:\windows\system32\cmd.EXE /D /E:ON /V:OFF /S /C "CALL "{0}""
  env:
    Qt5_Dir: d:/a/HippoPlayer/Qt/5.12.9/msvc2017_64
    QT_PLUGIN_PATH: d:/a/HippoPlayer/Qt/5.12.9/msvc2017_64/plugins
    QML2_IMPORT_PATH: d:/a/HippoPlayer/Qt/5.12.9/msvc2017_64/qml
CreateProcess() failed
errno: 0 (No error)
##[error]Process completed with exit code 1.
</code></pre>
<p>Running this locally works just fine. I have been trying to search the net for a solution but haven’t managed to fine one.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/emoon">@emoon</a>,</p>
<blockquote>
<p><strong>Running this locally works just fine.</strong></p>
</blockquote>
<p>Please try to install a self-hosted runner on the local machine, then use this self-hosted runner to run the “<strong>build_windows</strong>” job to see if the problem still exists.<br>
In this way, you do not need the “<strong>Install Qt</strong>” step in the workflow if you have installed Qt on the local machine.</p>
<p>Not sure if the “<strong>jurplel/install-qt-action@v2</strong>” action has setup the completed required Qt environments on the runner.</p>

**Comment by emoon** [comment]
<p>Thanks for the reply.</p>
<p>I can test a local runner. Right now I simply tried to remove the Qt install now and I get the same issue with the remote (default) runner. The script looks like this now</p>
<pre><code class="lang-auto">  build_windows:
    runs-on: windows-latest
    steps:
    - uses: actions/checkout@v2
#    - name: Install Qt
#      uses: jurplel/install-qt-action@v2
#    - name: Env test
#      run: |
#        echo "Testing"
#        echo %Qt5_DIR%
#        echo "Testing 2"
#        set QT5_BIN=%Qt5_DIR%/bin
#        echo %QT5_BIN%
    - name: Build Release
      shell: cmd
      run: bin\windows\tundra2.exe win64-msvc-debug
</code></pre>
<p>So all it’s doing is to checkout the repo and try to run the executable and with the same result. I will likely try to just build the executable used from source instead and see if it works then.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/emoon">@emoon</a>,</p>
<p>It’s strange that the command “<strong>bin\windows\tundra2.exe win64-msvc-debug</strong>” can work fine if you directly run it on your local machine that has Qt installed, but does not work when running on the self-hosted runner installed on this machine.</p>
<p>There may be some issues occur when run this command via the runner application.<br>
I have create an internal issue ticket to report this problem to the appropriate engineering team for further investigation and evaluation. If they have any progress, I will notify you in time, and sometimes the appropriate engineers may directly reply you here.</p>

**Comment by emoon** [comment]
<p>Thanks!</p>
<p>Also notice that this application doesn’t require Qt to be installed. It do rely on scripts and files that are included in the repository but if something of those should be missing the program should report it. Right now it seems like Windows can’t run the executable at all for some reason.</p>

**Comment by emoon** [comment]
<p>Hey, some more updates here: It actually turns out that the <code>tundra2.exe</code> executable gets started and the <code>CreateProcess failed()</code> gets reported by the program itself.</p>
<p>I tracked it down to this code</p>
<pre><code class="lang-auto">    if (!CreateProcessA(NULL, GetCommandLineA(), NULL, NULL, TRUE, CREATE_BREAKAWAY_FROM_JOB|CREATE_NEW_PROCESS_GROUP|CREATE_SUSPENDED, NULL, NULL, &amp;startup_info, &amp;proc_info))
      CroakErrno("CreateProcess() failed for %s error %d", GetCommandLineA(), GetLastError());
</code></pre>
<p>And I added some more info now and the output is</p>
<pre><code class="lang-auto">CreateProcess() failed for "D:\a\HippoPlayer\HippoPlayer\src\external\tundra\vs2017\x64\Release\tundra2.exe" error 5
</code></pre>
<p>(I have added the source code to the repo so I can do changes to it) the 5 is the error code from <code>GetLastError()</code> which is</p>
<pre><code class="lang-auto">ERROR_ACCESS_DENIED

    5 (0x5)

    Access is denied.
</code></pre>
<p>So for some reason the code above isn’t allowed create a new process. I will speak with the author of this code, but I still find it strange it doesn’t run correct on the builder instance.</p>

**Comment by emoon** [comment, marked as answer]
<p>And another update:</p>
<p>So now have a work-around for the problem. We aren’t actually fully sure but it might that the VMs are configured in such way that creating a process with <code>CREATE_NEW_PROCESS_GROUP</code> isn’t supported perhaps for some security reason.</p>
<p>In this case the code that does this isn’t actually needed when running on a CI fashion as it’s only there for handling aborting a build (ctrl+c) which doesn’t happen in this here.</p>

**Comment by al-cheb** [comment]
<p>Hello, <a class="mention" href="/emoon">@emoon</a><br>
I have created a bug - <a href="https://github.com/actions/runner/issues/595" rel="nofollow noopener">https://github.com/actions/runner/issues/595</a></p>

**Comment by emoon** [comment]
<p>Thanks! Great bug report :slight_smile:</p>
